elastic/logstash
# Remove bin/system-install (pleaserun) usage

## What does this PR do?

Remove the use of bin/system-install (which effectively uses pleaserun) to simplify package maintenance across Linux distributions. This script runs on package post installation, which creates system files (systemd/sysv) on the fly.

This makes it difficult to maintain, make changes, or fix issues that simply require to make minor changes on the init scripts/systemd units.

This PR, effectively does:
 - Removes pleaserun usage.
 - Adds a systemd unit file. (logstash.service)
    - Installs the unit under /lib/systemd/system
    - Removes the unit from the old location if present (/etc/systemd/system)
 - Adds a init.d file.
 - Adds a default file
 - Installs all files above (following what the *beats, ES and Kibana do).

NOTE; A follow-up PR will remove the actual Dependency from Logstash.



## Why is it important/What is the impact to the user?

It is important because it completely simplifies the maintenance of files for packaging. It also ensures that files (specifically the systemd unit) are installed where they belong.


## Checklist




## How to test this PR locally

To test:

1. Build the package:

```
$ sudo apt-get install openjdk-8-jdk
$ export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
$ curl -sSL https://get.rvm.io | bash -s stable --ruby
$ source /home/ubuntu/.rvm/scripts/rvm
$ rvm install jruby-9.2.13.0
$ rvm --default use jruby-9.2.13.0
$ ./gradlew installDefaultGems
$ rake artifact:deb
```

2. Install the created artifact (deb/rpm)
3. Ensure it has a systemd file, default file and/or sysv file.



## Related Issues

 - Closes https://github.com/elastic/logstash/issues/12439
 - Closes https://github.com/elastic/logstash/issues/10519
 - Closes https://github.com/elastic/logstash/issues/8253
